build.py

```diff
@@ -0,0 +1,221 @@
+    """
+    # this may fail with exceptions
+    cmd = 'git', 'describe', '--tags', '--long', '--dirty',
+    version = subprocess.check_output(cmd, stderr=subprocess.STDOUT).strip()
+    dirty = version.endswith('-dirty')
+    tag, distance, commit = version.split('-')[:3]
+    # lower tag and strip V prefix in tags
+    tag = tag.lower().lstrip('v ').strip()
+    # strip leading g from git describe commit
+    commit = commit.lstrip('g').strip()
+    return tag, int(distance), commit, dirty
+
+
+def create_zip(base_dir, archive_base_name,):
+    """
+    Create a zip at path `archive_name` from the files in `base_dir`
+    """
+    len_base_dir = len(base_dir) + 1
+
+    with zipfile.ZipFile(os.path.join(base_dir, archive_base_name + '.zip'), 'w', zipfile.ZIP_DEFLATED) as zipf:
+        for dirname, _, files in os.walk(os.path.join(base_dir, archive_base_name)):
+            relative_dir = dirname[len_base_dir:]
+            for name in files:
+                member_location = os.path.join(dirname, name)
+                member_path = os.path.join(relative_dir, name)
+                zipf.write(member_location, member_path)
+
+
+def create_tar(base_dir, archive_base_name):
+    """
+    Create a zip at path `archive_name` from the files in `base_dir`
+    """
+    len_base_dir = len(base_dir) + 1
+
+    with closing(tarfile.open(os.path.join(base_dir, archive_base_name + '.tar.gz'), mode='w:gz')) as tarf:
+        for dirname, _, files in os.walk(os.path.join(base_dir, archive_base_name)):
+            relative_dir = dirname[len_base_dir:]
+            for name in files:
+                member_location = os.path.join(dirname, name)
+                member_path = os.path.join(relative_dir, name)
+                tarf.add(member_location, member_path)
+
+
+def build(clean=True, app_name=APP_NAME,
+          platform=PLATFORM, platform_name=PLATFORM_NAME,
+          arch=ARCH, build_dir=BUILD_DIR, icon=ICON,
+          electron_version=ELECTRON_VERSION, asar=ASAR):
+    """
+    Run a build.
+    """
+    build_version = get_version()
+
+    print('=> BUILDING AboutCode App release:', build_version)
+
+    if clean:
+        # cleanup of previous build
+        shutil.rmtree(build_dir)
+        if not os.path.exists(build_dir):
+            os.makedirs(build_dir)
+
+    electron_args = [
+        '.',
+        app_name,
+        '--prune',
+        '--ignore=thirdparty/*',
+        '--ignore=dist/*',
+        '--ignore=.idea' ,
+        '--ignore=.gitignore' ,
+        '--ignore=test',
+        '--ignore=bower.json',
+        '--platform=' + platform,
+        '--arch=' + arch,
+        '--icon=assets/app-icon/' + icon,
+        '--version=' + electron_version,
+        '--out=' + build_dir,
+         '--asar=' + asar,
+        '--overwrite=true'
+    ]
+
+    # find the path to the NPM bin directory
+    npm_bin = subprocess.check_output(['npm', 'bin'], stderr=subprocess.STDOUT).strip()
```

Comment:
This seems to be failing on Windows. It might need a `shell=True` flag there (this is a common Windows issue for Python subprocesses)